### PR TITLE
Fix RT-1.3: BGP route propagation needs NI type, router-id and peer-group

### LINKED ISSUE #312: Bug RT-1.3: BGP route propagation needs NI type, router-id and peer-group
Hi,
BGP on Juniper requires network instance type, router-id and peer-group to be configured.  Script needs to be updated with same.

Refs:

(https://github.com/openconfig/featureprofiles/blob/1b105fe0d3bdf7f866dc52aa15e4ea5327be01c1/yang/fpoc/path_structs-0.go#L9810)
`func NetworkInstanceAny()`

(https://github.com/openconfig/featureprofiles/blob/1b105fe0d3bdf7f866dc52aa15e4ea5327be01c1/yang/fpoc/path_structs-1.go#L10204)
`func RouterId() `

(https://github.com/openconfig/featureprofiles/blob/1b105fe0d3bdf7f866dc52aa15e4ea5327be01c1/yang/fpoc/structs-1.go#L13104)
`func GetOrCreatePeerGroup()`

Thanks.

### PR DESCRIPTION
**Pull request by yashikakn**
Hi,
Resolves #312. Please review and provide approval. 

Thanks.

### CODE REVIEW

File: feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go [hunk 1]
```diff
@@ -122,8 +122,8 @@ func (d *dutData) Configure(t *testing.T, dut *ondatra.DUTDevice) {
 		dut.Config().Interface(ocName).Replace(t, a.NewInterface(ocName))
 	}
 	dutBGP := dut.Config().NetworkInstance(*deviations.DefaultNetworkInstance).
-		Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
-	dutBGP.Replace(t, d.bgpOC)
+		Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()	
+	dutBGP.Replace(t, d.bgpOC)
```
Comment (liulk) [inline feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go:126]: I don't see what's changed here besides the redundant whitespace?

File: feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go [hunk 2]
```diff
@@ -156,10 +156,12 @@ func TestBGP(t *testing.T) {
 				"192.0.2.2": {
 					PeerAs:          ygot.Uint32(ateAS1),
 					NeighborAddress: ygot.String("192.0.2.2"),
+					PeerGroup: ygot.String("BGP-PEER-GROUP1"),
```
Comment (liulk) [inline feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go:159]: Run gofmt?

File: feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go [hunk 3]
```diff
@@ -276,6 +282,12 @@ func TestBGP(t *testing.T) {
 			}
 
 			dut := ondatra.DUT(t, "dut")
+
+			t.Log("Configure Network Instance")
```
Comment (liulk) [inline feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go:286]: This is supposed to be done inside tc.dut.Configure() if not already.

**Review by liulk — CHANGES_REQUESTED** [review]

File: feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go (hunk 2 above)
Comment (yashikakn) [reply feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go]: Done. Thanks.

File: feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go (hunk 1 above)
Comment (yashikakn) [reply feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go]: Fixed it. Thanks.

File: feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go (hunk 3 above)
Comment (yashikakn) [reply feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go]: Moved network instance config to  tc.dut.Configure(). Thanks.

File: feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go [hunk 4]
```diff
@@ -121,6 +121,12 @@ func (d *dutData) Configure(t *testing.T, dut *ondatra.DUTDevice) {
 		ocName := dut.Port(t, a.Name).Name()
 		dut.Config().Interface(ocName).Replace(t, a.NewInterface(ocName))
 	}
+
+	t.Log("Configure Network Instance")
+	dutConfNIPath := dut.Config().NetworkInstance(*deviations.DefaultNetworkInstance)
+	dutConfNIPath.Type().Replace(t, telemetry.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+	dutConfNIPath.RouterId().Replace(t, dutPort2.IPv4)
+
```
Comment (singh-prem) [inline feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go:129]: use BGP router id as in https://github.com/openconfig/featureprofiles/pull/567

Comment (yashikakn) [reply feature/bgp/addpath/ate_tests/route_propagation_test/route_propagation_test.go:128]: Done. Thanks.

**Review by bstoll — APPROVED** [review]
Thanks, this LGTM.

I am not sure why we need to configure the default instance type on the default network instance (line 127).  This seems like it should be already set.  I have seen this pattern coming up in other tests and I think we either need to clarify why this is required or create a deviation/bug for this pattern.

**Review by shichuzhu — APPROVED** [review]
Good for Go.

**Comment by yashikakn** [comment]
@liulk : Could you please help merge this PR. All the review comments have been addressed.
Thanks.

**Review by liulk — APPROVED** [review]

**Pull request merged**